x-pack/solutions/observability/plugins/observability_ai_assistant_app/scripts/evaluation/read_kibana_config.ts

```diff
@@ -35,9 +35,9 @@ export const readKibanaConfig = () => {
   };
 
   return {
-    'elasticsearch.hosts': 'http://localhost:9200',
-    'elasticsearch.username': 'elastic',
-    'elasticsearch.password': 'changeme',
+    'elasticsearch.hosts': get(loadedKibanaConfig, 'elasticsearch.hosts', 'http://localhost:9200'),
```

Comment:
why are these necessary btw? everything from `loadedKibanaConfig` should override on L41

Reply:
It's necessary because sometimes in kibana.dev.yml `elasticsearch` keys are configured nested versus dot notation.

The original implementation worked when everything was configured with `elasticsearch.x` config. However, many configurations, including the configuration provided by oblt-cli, prefer nested elasticsearch keys.

Reply:
hmm I suppose I would have expected js-yaml to take care of this. Can you instead use `set` to "unflatten" the entire object (rather than selectively picking props from it)?

Reply:
https://github.com/elastic/kibana/blob/1e98a368188aab5b120db7f012754af386db77bd/x-pack/solutions/observability/packages/utils_common/object/unflatten_object.ts#L10-L11

Reply:
apologies, I mean unflatten the statically defined keys here and unflatten the config